### PR TITLE
🎉 CDK: Add support for custom headers passing to the request in `OAuth2Authenticator. refresh_access_token`

### LINKED ISSUE #4016: Allow headers to be added in the request for refreshing access token
## Tell us about the problem you're trying to solve
Regarding the connector development, I'm using the [OAuth2Authenticator](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/streams/http/auth/oauth.py) and I see that there's currently no way to add header(s) to the [token refresh call](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/streams/http/auth/oauth.py#L79).

The source I'm working on requires an authorization header (Basic token) to be able to send an authorized header to their authorization server.

## Describe the solution you’d like
It's good if the developers have an option to attach an optional header as the OAuth2Authenticator is instantiated--which will be used in the `refresh_access_token` method.

## Describe the alternative you’ve considered or used
No alternatives so far except from modifying the OAuth2Authenticator class to accept headers.

### PR DESCRIPTION
**Pull request by Zirochkaa**
Closes #4016.

### CODE REVIEW

**Comment by yevhenii-ldv** [comment]
Why did you decide to add so many __init__ files to this PR?

**Comment by Zirochkaa** [comment]
> Why did you decide to add so many **init** files to this PR?

Because without those init files I can't run tests in a specific file. For example I can't run following command:
```
pytest -s unit_tests/sources/streams/http/auth
```

So that's why I decided to add init file to all test folders so anybody can run tests in a specific file and not just using command `pytest -s unit_tests`.

**Review by avida — APPROVED** [review]

**Review by yevhenii-ldv — APPROVED** [review]

**Review by sherifnada — COMMENTED** [review]
please add unit tests

**Review by htrueman — APPROVED** [review]

File: airbyte-cdk/python/unit_tests/sources/streams/http/auth/test_auth.py [hunk 1]
```diff
@@ -130,18 +129,22 @@ def test_refresh_request_body(self):
         }
         assert body == expected
 
-    def test_refresh_access_token(self, mocker):
+    def test_refresh_access_token(self, requests_mock):
```
Comment (sherifnada) [inline airbyte-cdk/python/unit_tests/sources/streams/http/auth/test_auth.py:132]: looks great!

**Review by sherifnada — APPROVED** [review]

**Pull request merged**